### PR TITLE
New - Image block

### PR DESCRIPTION
**Pull request by imrantushar**
### Description
New Block for an image. This Block is only for showcasing images with some necessary info(title, caption)

### Types of changes
New Image Block Added
New SelectControl Components Added

### How has this been tested?
This is just another simple block. Used to showcase just image. Here is some test case below

Multi-Type Image Upload (direct upload, default uploader, upload from URL)
Check all toolbar settings like alignment, duotone filter, cropping, etc
Check all general & styles settings
Default and Custom mask settings with mask position

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: includes/blocks/image/frontend.css.php [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+/**
+ * Frontend CSS & Google Fonts loading File.
+ *
+ * @since 2.0.0
+ *
+ * @package uagb
+ */
+
+// Add fonts.
+UAGB_Block_JS::blocks_advanced_heading_gfont( $attr );
```
Comment (RajkiranBagal) [inline includes/blocks/image/frontend.css.php:11]: Do we need this here?

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: includes/blocks/image/frontend.css.php (hunk 1 above)
Comment (imrantushar) [reply includes/blocks/image/frontend.css.php:11]: @RajkiranBagal yes we need it but it is the wrong method name. I am fixing that

**Review by RajkiranBagal — APPROVED** [review]

File: blocks-config/image/class-uagb-image.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		 * @since 1.18.1
+		 * @var instance
+		 */
+		private static $instance;
+
+
+		/**
+		 *  Initiator
+		 *
+		 * @since 1.18.1
```
Comment (AbhijitNage123) [inline blocks-config/image/class-uagb-image.php:32]: since x.x.x

File: blocks-config/image/class-uagb-image.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+		 * Constructor
+		 */
+		public function __construct() {
+			add_action( 'init', array( $this, 'register_blocks' ) );
+		}
+
+		/**
+		 * Registers the `core/latest-posts` block on server.
+		 *
+		 * @since 0.0.1
```
Comment (AbhijitNage123) [inline blocks-config/image/class-uagb-image.php:51]: since x.x.x

File: src/blocks/image/block.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import UAGB_Block_Icons from '@Controls/block-icons';
+import { __ } from '@wordpress/i18n';
+import './style.scss';
+import { registerBlockType } from '@wordpress/blocks';
+
+
+registerBlockType( 'uagb/image', {
+	title: __( 'Advanced Image', 'ultimate-addons-for-gutenberg' ),
+	description: __( 'This block let\'s you add an Image.', 'ultimate-addons-for-gutenberg' ),
+	icon: UAGB_Block_Icons.post_masonry,
```
Comment (AbhijitNage123) [inline src/blocks/image/block.js:17]: We need an image Icon here. Get in touch with the designer

File: src/blocks/image/edit.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+		setAttributes( { block_id: props.clientId.substr( 0, 8 ) } );
+
+	}, [] );
+
+	useEffect( () => {
+		// Replacement for componentDidUpdate.
+		const blockStyling = styling( props );
+
+        addBlockEditorDynamicStyles( 'uagb-image-style-' + props.clientId.substr( 0, 8 ), blockStyling );
+	}, [ props ] );
```
Comment (AbhijitNage123) [inline src/blocks/image/edit.js:36]: I think we need to pass deviceType too. So CSS style tag will generate on responsive devices. Please check once

**Review by AbhijitNage123 — CHANGES_REQUESTED** [review]

File: blocks-config/image/class-uagb-image.php [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+		/**
+		 * Constructor
+		 */
+		public function __construct() {
+			add_action( 'init', array( $this, 'register_blocks' ) );
+		}
+
+		/**
+		 * Registers the `core/latest-posts` block on server.
```
Comment (AbhijitNage123) [inline blocks-config/image/class-uagb-image.php:49]: * Register the `image` block on the server.

**Review by AbhijitNage123 — CHANGES_REQUESTED** [review]

File: blocks-config/image/class-uagb-image.php (hunk 2 above)
Comment (imrantushar) [reply blocks-config/image/class-uagb-image.php:32]: Fixed

File: blocks-config/image/class-uagb-image.php (hunk 3 above)
Comment (imrantushar) [reply blocks-config/image/class-uagb-image.php:51]: Fixed

File: src/blocks/image/block.js (hunk 4 above)
Comment (imrantushar) [reply src/blocks/image/block.js:17]: Already talked with designers and waiting...

File: blocks-config/image/class-uagb-image.php (hunk 6 above)
Comment (imrantushar) [reply blocks-config/image/class-uagb-image.php]: Updated comment

**Review by AbhijitNage123 — APPROVED** [review]

**Pull request merged**